usaspending_api/etl/tests/integration/test_swap_in_new_table.py

```diff
@@ -0,0 +1,189 @@
+        cursor.execute(
+            "ALTER TABLE test_table_temp DROP CONSTRAINT test_table_wrong_col_1_constraint_temp;"
+            "ALTER TABLE test_table_temp ADD CONSTRAINT test_table_col_1_constraint_temp CHECK (col1 != 'TEST_WRONG');"
+        )
+        try:
+            call_command("swap_in_new_table", "--table=test_table")
+        except Exception as e:
+            assert str(e) == (
+                "The constraint definitions are different for the tables: test_table_temp and test_table."
+            )
+        else:
+            assert False, "No exception was raised"
+
+
+@mark.django_db()
+def test_column_validation():
+    with connection.cursor() as cursor:
+        cursor.execute(
+            "CREATE TABLE test_table (col1 TEXT, col2 INT);"
+            "CREATE TABLE test_table_temp (col1 TEXT, col2 INT, col3 INT);"
+        )
+        try:
+            call_command("swap_in_new_table", "--table=test_table")
+        except Exception as e:
+            assert str(e) == (f"The number of columns are different for the tables: test_table_temp and test_table.")
+        else:
+            assert False, "No exception was raised"
+
+        cursor.execute(
+            "DROP TABLE test_table;"
+            "DROP TABLE test_table_temp;"
+            "CREATE TABLE test_table (col1 TEXT, col2 INT);"
+            "CREATE TABLE test_table_temp (col1 TEXT, col2 TEXT);"
+        )
+        try:
+            call_command("swap_in_new_table", "--table=test_table")
+        except Exception as e:
+            assert str(e) == (f"The column definitions are different for the tables: test_table_temp and test_table.")
+        else:
+            assert False, "No exception was raised"
+
+
+@mark.django_db()
+def test_happy_path():
+    with connection.cursor() as cursor:
+        cursor.execute(
+            "CREATE TABLE test_table (col1 TEXT, col2 INT);"
+            "CREATE TABLE test_table_temp (col1 TEXT, col2 INT);"
+            "INSERT INTO test_table (col1, col2) VALUES ('goodbye', 1);"
+            "INSERT INTO test_table_temp (col1, col2) VALUES ('hello', 2), ('world', 3);"
+            "CREATE INDEX test_table_col1_index ON test_table(col1);"
+            "CREATE INDEX test_table_col1_index_temp ON test_table_temp(col1);"
+            "ALTER TABLE test_table ADD CONSTRAINT test_table_col_1_constraint CHECK (col1 != 'TEST');"
+            "ALTER TABLE test_table_temp ADD CONSTRAINT test_table_col_1_constraint_temp CHECK (col1 != 'TEST');"
+        )
+        call_command("swap_in_new_table", "--table=test_table")
+        cursor.execute("SELECT * FROM test_table ORDER BY col1")
+        result = ordered_dictionary_fetcher(cursor)
+        assert result == [{"col1": "hello", "col2": 2}, {"col1": "world", "col2": 3}]
+
+        cursor.execute(
+            "SELECT * FROM information_schema.tables WHERE table_name IN ('test_table_temp', 'test_table_old')"
+        )
+        result = cursor.fetchall()
+        assert len(result) == 0
```

Comment:
Could you create an almost identical happy path with the `--keep-old-data` flag?